atoMEC/mathtools.py

```diff
@@ -194,6 +239,46 @@ def fd_int_complete(mu, beta, n):
     return I_n
 
 
+def ideal_entropy_int(mu, beta, n):
+    r"""
+    Compute the entropy for the ideal electron gas (without prefactor) - see notes.
+
+    Parameters
+    ----------
+    mu : float
+        chemical potential
+    beta: float
+        inverse temperature
+    n : int
+        order of Fermi-Dirac integral (see notes)
+
+    Returns
+    -------
+    I_n : float
+        the complete fermi-dirac integral
+
+    Notes
+    -----
+    The entropy of an ideal electron gas is defined as
+
+    .. math::
+        I_n(\mu,\beta) = \int \mathrm{d}\epsilon \epsilon^{n/2}
+        (f_\mathrm{fd}\log{f_\mathrm{fd}} + (1-f_\mathrm{fd}) \log(1-f_\mathrm{fd}) ),
+
+    where :math:`f_\mathrm{fd}=f_\mathrm{fd}(\epsilon,\mu,\beta)` is the Fermi-Dirac
+    distribution.
+    """
+    # use scipy quad integration routine
+    limup = np.inf
+
+    # ignore integration warnings (omnipresent because of inf upper limit)
+    with warnings.catch_warnings():
+        warnings.filterwarnings("ignore")
```

Comment:
I am not sure about this. Suppress warnings only when there is a good reason for that. For which parameter range `quad` actually raises these? If this is due to singularities within the integrand, then specify these with the `points` parameter. 

I also recommend adding tests in general to test expected outcome for a specific parameter set (including edge cases).

Reply:
Fine by me to do it this way. Just to note that there is also the same warning suppression in the `fd_int_complete` function. 

So my question is, should we deal only with the newly introduced warning in the current PR, or both? Or, merge this PR as is and then open a new one to address both issues simultaneously?

![Screenshot from 2021-07-26 09-23-36](https://user-images.githubusercontent.com/18050753/126949109-29b016cb-5e00-4ee9-9e51-3158a05802e8.png)

Reply:
Well, then lets merge and open up a new issue on the warnings.